tests/fsharp/Compiler/ErrorMessages/SuggestionsTests.fs

```diff
@@ -0,0 +1,410 @@
+    let ``Suggest Methods`` () =
+        CompilerAssert.TypeCheckSingleError
+            """
+module Test2 = 
+    type D() = 
+
+       static let x = 1
+
+       member x.Method1() = 10
+
+    D.Method2()
+
+exit 0
+            """
+            FSharpErrorSeverity.Error
+            39
+            (9, 17, 9, 24)
+            "The field, constructor or member 'Method2' is not defined.\nMaybe you want one of the following:\nMethod1"
+
+
+
+    [<Test>]
+    let ``Suggest Modules`` () =
+        CompilerAssert.TypeCheckSingleError
+            """
+module Collections =
+
+    let f () = printfn "%s" "Hello"
+
+open Collectons
+
+exit 0
+            """
+            FSharpErrorSeverity.Error
+            39
+            (6, 6, 6, 16)
+            "The namespace or module 'Collectons' is not defined\nMaybe you want one of the following:\nCollections"
+
+
+    [<Test>]
+    let ``Suggest Namespaces`` () =
+        CompilerAssert.TypeCheckSingleError
+            """
+open System.Collectons
+
+exit 0
+            """
+            FSharpErrorSeverity.Error
+            39
+            (2, 13, 2, 23)
+            "The namespace 'Collectons' is not defined\nMaybe you want one of the following:\nCollections"
+
+
+    //TODO This was marked with SCFLAGS="--vserrors" in tests/fsharpqa/Source/Warnings/env.lst file
+    [<Test>]
+    let ``Suggest Record Labels`` () =
+        CompilerAssert.TypeCheckSingleError
+            """
+type MyRecord = { Hello: int; World: bool}
+
+let r = { Hello = 2 ; World = true}
+
+let x = r.ello
+
+exit 0
+            """
+            FSharpErrorSeverity.Error
+            39
+            (6, 11, 6, 15)
+            "The field, constructor or member 'ello' is not defined.\nMaybe you want one of the following:\s+Hello"
+
+
+    //TODO This was marked with SCFLAGS="--vserrors" in tests/fsharpqa/Source/Warnings/env.lst file
+    [<Test>]
+    let ``Suggest Record Type for RequireQualifiedAccess Records`` () =
+        CompilerAssert.TypeCheckSingleError
+            """
+[<RequireQualifiedAccess>]
+type MyRecord = {
+    Field1: string
+    Field2: int
+}
+
+let r = { Field1 = "hallo"; Field2 = 1 }
+
+exit 0
+            """
+            FSharpErrorSeverity.Error
+            39
+            (8, 11, 8, 17)
+            "The record label 'Field1' is not defined.\nMaybe you want one of the following:\s+MyRecord.Field1"
+
+
+    ///TODO Check if the second error message is correct
+    [<Test>]
+    let ``Suggest To Use Indexer`` () =
+        CompilerAssert.TypeCheckSingleError
+            """
+let d = [1,1] |> dict
+let y = d[1]
+
+let z = d[|1|]
+
+let f() = d
+let a = (f())[1]
+
+exit 0
+            """
+            [|
+                FSharpErrorSeverity.Error, 3217, (2, 10, 2, 13), "This value is not a function and cannot be applied. Did you intend to access the indexer via d.\[index\] instead?"
+                FSharpErrorSeverity.Error, 3, (5, 9, 5, 9), "This value is not a function and cannot be applied.$"
```

Comment:
This error message is correct?

Reply:
Probably not, take a look here https://github.com/dotnet/fsharp/pull/7226#issuecomment-511179308
Old tests pass when error messages contains string specified in `<Expects ...>`. In new test you need to specify full error message. 

P.S. Also you do not need `exit 0` in new tests.